### DISCUSSION TITLE
[help] Testing functions under R/

### DISCUSSION THREAD

**Discussion by asadow** [ropensci/targets#1176] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I encountered a situation where I had edited a function under R/ and without testing, it broke the pipeline. I debugged and then reverted the edit.

So, I need tests. I don't want to create a new package to test, however. I do believe the functions under R/ belong in the targets project.

Is the best course of action to create a testthat folder and during development run testthat::test_dir(".")?

**Comment by wlandau** [comment, marked as answer]
> Is the best course of action to create a testthat folder and during development run testthat::test_dir(".")?

If it works in your case, then by all means. The whole setup should be more flexible than if you were writing a formal R package (i.e. the test suite could be just a bunch of scripts in `tests/` instead of `tests/testthat/`).
